### ISSUE TITLE
[Internal] ocamldebug does not work anymore on semgrep Main.bc

### ISSUE THREAD

**Issue by aryx** [semgrep/semgrep#4083]
Get some Fatal error: debugger does not support channel locks

```
[pad@yrax yy (develop)]$ ./scripts/run-ocamldebug.sh ./_build/default/src/cli/Main.bc --help
                       
ocamldebug hints:
* info modules: List all program modules.  Helpful to get their names.
* break @ [<module>] <line>: Insert breakpoint at line.
  - NOTE: Always defaults to frame 0 module even after "up"!
  - Sometimes has to be placed on the preceding line to work.
* break [<module>.]<function>: Break at function.
  - Must run program first, e.g.: "r", "break ...", "g 0", "r", "up"... .
  - Useful breakpoint: OUnit.assert_failure (or Alcotest equivalent)
* r[un], s[tep], n[ext]: gdb-like program running.
* bt, frame, up, down: gdb-like call stack navigation.
* g[oto] <n>: Jump forward or backward in time.
* print <var>: Print a variable.
* fin[ish]: Run until current function returns.
* print *: Print the most recent function result (e.g., after finish).
* list [startline [endline]]: Print source code around current location.
* help [<command>]: List commands or info about one.
* quit: Exit (also Ctrl-D).

	OCaml Debugger version 4.12.0

run
Loading program... done.
Fatal error: debugger does not support channel locks
```

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-505/[internal]-ocamldebug-does-not-work-anymore-on-semgrep-mainbc.

**Comment by aryx** [comment]
it works!!!!

![image](https://user-images.githubusercontent.com/332336/173185285-4b822c31-ee12-4cc2-a44d-3244f7d7a70d.png)

**Comment by aryx** [comment]
This was fixed when we switch to OCaml 4.14, as Xavier himself fixed the issue in OCaml 4.13.2
